backend/accounting-service/src/main/kotlin/services/projects/v2/ProjectService.kt

```diff
@@ -1245,6 +1249,267 @@ class ProjectService(
+    }
+
+    suspend fun deleteInviteLink(
+        actorAndProject: ActorAndProject,
+        request: ProjectsDeleteInviteLinkRequest,
+        ctx: DBContext = db
+    ) {
+        val project = actorAndProject.requireProject()
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            val success = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", request.token)
+                },
+                """
+                    delete from project.invite_links where project_id = :project and token = :token
+                """
+            ).rowsAffected > 0
+
+            if (!success) {
+                throw RPCException(
+                    "Unable to delete invitation link",
+                    HttpStatusCode.BadRequest
+                )
+            }
+        }
+    }
+
+    suspend fun updateInviteLinkRoleAssignment(
+        actorAndProject: ActorAndProject,
+        request: ProjectsUpdateInviteLinkRoleAssignmentRequest,
+        ctx: DBContext = db
+    ) {
+        val project = actorAndProject.requireProject()
+
+        if (!listOf(ProjectRole.ADMIN, ProjectRole.USER).contains(request.role)) {
+            throw RPCException("Role assignment can only be either Admin or User", HttpStatusCode.BadRequest)
+        }
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            val success = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", request.token)
+                    setParameter("role", request.role.name)
+                },
+                """
+                    update project.invite_links
+                    set role_assignment = :role
+                    where project_id = :project and token = :token
+                """
+            ).rowsAffected > 0
+
+            if (!success) {
+                throw RPCException(
+                    "Unable to update role assignment",
+                    HttpStatusCode.BadRequest
+                )
+            }
+        }
+    }
+
+    suspend fun updateInviteLinkGroupAssignment(
+        actorAndProject: ActorAndProject,
+        request: ProjectsUpdateInviteLinkGroupAssignmentRequest,
+        ctx: DBContext = db
+    ) {
+        val project = actorAndProject.requireProject()
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", request.token)
+                    setParameter("groups", request.groups)
+                },
+                """
+                    with new (group_id, token) as (
```

Comment:
I would add more newlines here to be consistent with other SQL queries in UCloud/Core.